test/AutoRest.TestServer.Tests/Infrastructure/CustomClientRequestIdPolicy.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace AutoRest.TestServer.Tests.Infrastructure
+{
+    public class CustomClientRequestIdPolicy : HttpPipelineSynchronousPolicy
+    {
+        public CustomClientRequestIdPolicy()
+        {
+        }
+
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            base.OnSendingRequest(message);
+
+            if (ClientRequestIdScope.CurrentClientRequestId != null)
+            {
+                message.Request.ClientRequestId = ClientRequestIdScope.CurrentClientRequestId;
```

Comment:
So, why couldn't we do this before? This was what I was thinking originally. Basically, inject a 'replace value' line like this in the pipeline.